selftests/unit/test_future_settings.py

```diff
@@ -90,6 +90,19 @@ def test_add_argparser(self):
         stgs.add_argparser_to_option('section.key', parser, '--other-arg',
                                      allow_multiple=True)
 
+    def test_multiple_parsers(self):
+        stgs = settings.Settings(self.config_file.name)
+        parser1 = argparse.ArgumentParser(description='description')
+        stgs.register_option('section', 'key', 'default', 'help',
+                             parser=parser1, long_arg='--first-option')
+        parser2 = argparse.ArgumentParser(description='other description')
+        with self.assertRaises(settings.DuplicatedNamespace):
+            stgs.register_option('section', 'key', 'default', 'help',
+                                 parser=parser2, long_arg='--other-option')
+        stgs.register_option('section', 'key', 'default', 'help',
+                             parser=parser2, long_arg='--other-option',
+                             allow_multiple=True)
```

Comment:
Should we add there the assert to make sure that the second registration was successfully?

Reply:
> Should we add there the assert to make sure that the second registration was successfully?

The best I could come up with was:

```diff
diff --git a/selftests/unit/test_future_settings.py b/selftests/unit/test_future_settings.py
index 78dd90cc..bbab5b44 100644
--- a/selftests/unit/test_future_settings.py
+++ b/selftests/unit/test_future_settings.py
@@ -102,6 +102,7 @@ class SettingsTest(unittest.TestCase):
         stgs.register_option('section', 'key', 'default', 'help',
                              parser=parser2, long_arg='--other-option',
                              allow_multiple=True)
+        self.assertIs(stgs._namespaces['section.key'].parser, parser2)
 
     def tearDown(self):
         os.unlink(self.config_file.name)
```

But it uses private attributes.  Do you have other suggestion?

Reply:
That is good.